kernel/src/fs/sysfs/test.rs

```diff
@@ -0,0 +1,625 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{
+    borrow::Cow,
+    collections::BTreeMap,
+    format,
+    string::{String, ToString},
+    sync::{Arc, Weak},
+    vec,
+    vec::Vec,
+};
+use core::{any::Any, fmt::Debug};
+
+use ostd::{
+    mm::{FallibleVmRead, FallibleVmWrite, VmReader, VmWriter},
+    prelude::ktest,
+};
+use spin::RwLock;
+use systree::{
+    init_for_ktest, singleton as systree_singleton, Error as SysTreeError, Result as SysTreeResult,
+    SysAttrFlags, SysAttrSet, SysAttrSetBuilder, SysBranchNode, SysNode, SysNodeId, SysNodeType,
+    SysObj, SysStr, SysSymlink, SysTree,
+};
+
+use crate::{
+    fs::{
+        sysfs::fs::SysFs,
+        utils::{DirentVisitor, FileSystem, InodeMode, InodeType},
+    },
+    time::clocks::init_for_ktest as time_init_for_ktest,
+    Result,
+};
+
+// --- Mock SysTree Components ---
+// Sysfs acts as a view layer over the systree component.
+// These mocks simulate the systree interface (SysNode, SysBranchNode, etc.)
+
+// Mock SysNode for testing attributes
+#[derive(Debug)]
+struct MockLeafNode {
+    id: SysNodeId,
+    name: SysStr,
+    attrs: SysAttrSet,
+    data: RwLock<BTreeMap<String, String>>, // Store attribute data
```

Comment:
Could you try to reuse the helper types from `systree::utils`  so that the unit tests of systree and sysfs can be implemented easier, with less boilerplate code?